### ISSUE TITLE
Marking a message in mu4e-headers mode leads to Emacs freeze

### ISSUE THREAD

**Issue by acacar** [djcb/mu#1281]
## Expected or desired behavior

Message should be marked for refiling, read/unread, flag etc. 

## Actual behavior

Emacs hangs with %100 CPU utilization on a single core. Cannot be canceled with C-g, cannot be stopped with anything short of a kill -9. Same for GUI or console emacs.

## Versions of mu, mu4e/emacs, operating system etc.

GNU Emacs 25.3 and 26.1; mu 9.18 and 1.0; all combinations thereof
OS: Kubuntu 18.04 (KDE 12.5.6)
MTA: isync
sendmail: msmtp

## Any other detail

This is really weird. I have been using mu and mu4e happily -- *thank you very much for this beautiful piece software, by the way* -- for the last year and a half. Suddenly, today, it started freezing Emacs any time I try to mark anything in the headers view (e.g. archiving from Inbox). I've tried marking for refile, marking unread/read, flagging, and they cause the freeze immediately. Deferred mark (*) works until you select an action and then it freezes. Mark for move (with 'm') is working correctly.

I've tried two versions of emacs (25.3 and 26.1) as well as two versions of mu ( 9.18 and 1.0 ), in all four combinations and results are the same. The fact that this happened out of the blue today without any software updates leads me to believe that something changed in my maildir that mu/mu4e didn't like. I know this is probably something to do with my configuration/state rather than mu4e but I would really appreciate it if you could give me some pointers to debug this. The relevant part of my init.el is as follows:

```
(use-package mu4e
  :bind (("C-c m" . mu4e))
  :config
  (setq mu4e-maildir "~/Maildir"
        mu4e-trash-folder "/Trash"
        mu4e-refile-folder "/Archive"
        mu4e-sent-folder "/Sent"
        mu4e-get-mail-command "mbsync fastmail"
        mu4e-update-interval nil
        mu4e-view-show-addresses t
        mu4e-view-show-images nil
        mu4e-drafts-folder "/Drafts"
        mu4e-change-filenames-when-moving t
        mu4e-attachment-dir  "~/Downloads"
        mu4e-compose-format-flowed t
        mu4e-html2text-command "w3m -dump -T text/html"
        mu4e-headers-include-related t
        )
  (when (fboundp 'imagemagick-register-types)
    (imagemagick-register-types))
  (add-to-list 'mu4e-view-actions
               '("View in browser" . mu4e-action-view-in-browser) t)
  (setq message-send-mail-function 'message-send-mail-with-sendmail
        sendmail-program "msmtp")) 
```

**Comment by acacar** [comment]
Had more time to debug this today. It was my fault: I forgot that I had installed the [FiraCode](https://github.com/tonsky/FiraCode) font and added ligature emulation to my .emacs (with [this workaround](https://github.com/tonsky/FiraCode/wiki/Emacs-instructions)). 

Apparently that is what causes the hang when mu4e tries to overlay destination information (e.g. `-> "/Archive"`) next to the header that has just been marked. If the overlay string has a sequence that needs to be composed into a ligature (in this case `->`) the freeze happens. That is as far as I went. I disabled the ligature emulation (didn't like it anyway), so I'm closing the issue. 

I'm leaving this comment just in case someone else has a similar problem. The freeze happens on the C source part of Emacs in the built-in function `overlay-put` called [here](https://github.com/djcb/mu/blob/4f6fd873fa82e899a03347321dc353e6ddbbac1a/mu4e/mu4e-mark.el#L282)
